webapp/src/modules/auction/sagas.js

```diff
@@ -71,3 +90,49 @@ function* handleBidRequest(action) {
     yield put(bidOnParcelsFailure(error.message))
   }
 }
+
+function* handleFetchAuctionRateRequest(action) {
+  const { token } = action
+  const selectedToken = yield select(state => getSelectedToken(state))
+  if (selectedToken !== token) {
+    yield put(navigateTo(locations.auction(token)))
+  }
+  try {
+    const { currentPrice } = yield select(getParams)
+    const landAuction = eth.getContract('LANDAuction')
+
+    let rate = 1
+    if (TOKENS.MANA !== TOKENS[selectedToken]) {
+      rate = eth.utils.fromWei(
+        yield call(() =>
+          landAuction.getRate(
+            TOKENS.MANA,
+            TOKENS[selectedToken],
+            eth.utils.toWei(currentPrice)
+          )
+        )
+      )
+    }
+    yield put(fetchAuctionRateSuccess(selectedToken, rate))
+  } catch (error) {
+    yield put(fetchAuctionRateFailure(selectedToken, error.message))
+  }
+}
+
+function* handleFetchAuctionRateSuccess(action) {
+  const token = yield select(state => getSelectedToken(state))
```

Comment:
Just to be consistent
```suggestion
  const token = yield select(getSelectedToken)
```

Reply:
✅ done